From 3fba12059a425d536aab88f440d2395c56f8d8bb Mon Sep 17 00:00:00 2001
From: Andreas Schmidt <andreas.schmidt@wago.com>
Date: Fri, 8 Dec 2017 09:33:54 +0100
Subject: [PATCH 133/261] dts: pfc200v3: enable pullup for GPIO uart5.rx

In PFC200v3 UART5 RX will driven by KBus CPU, but not till first communication.
Till first communication UART5 RX could be low (after reset state of GPIO is
INPUT with PULLDOWN).
At first time communication KBus CPU drive RX (his TX) pin to high and start
communication nearly immediately (after about 5us). Sometimes first received character
on mail CPU (AM335X) will be interpret wrong, because start bit was miss. To avoid this
pullup will enable while initialisation for UART5 pin muxing.
---
 arch/arm/boot/dts/wago-pfc200v3.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/wago-pfc200v3.dtsi b/arch/arm/boot/dts/wago-pfc200v3.dtsi
index ad9ba9c..6567821 100644
--- a/arch/arm/boot/dts/wago-pfc200v3.dtsi
+++ b/arch/arm/boot/dts/wago-pfc200v3.dtsi
@@ -189,7 +189,7 @@
 		 pinctrl-single,pins = <
 			/*  used for (XE164 KBUS image download or cortex-m0 image download) */
 			0x118 (PIN_OUTPUT | SLEWCTRL_SLOW | MUX_MODE3) /* mii1_rx_dv.uart5.txd */	//j17 TXD-XE164-CORTEX
-			0x0D8 (PIN_INPUT | MUX_MODE4) /* lcd_data14.uart5.rxd */	//v4  RXD-XE164-CORTEX
+			0x0D8 (PIN_INPUT_PULLUP | MUX_MODE4) /* lcd_data14.uart5.rxd */	//v4  RXD-XE164-CORTEX
 		 >;
 	};
 
-- 
2.7.4

